scripts/automation/create_release_tag.py

```diff
@@ -0,0 +1,446 @@
+        match = re.search(version_pattern, history_content, re.DOTALL | re.MULTILINE)
+
+        if match:
+            return match.group(1).strip()
+
+        return "No history notes found in source code"
+
+    except Exception as e:
+        print(f"Error reading history from source: {e}")
+        return "No history notes found for this version"
+
+
+def main():
+    try:
+        # Get added lines from git diff
+        added_lines = get_file_diff()
+        if not added_lines:
+            print("No changes found in index.json")
+            return
+
+        # Parse filenames from added lines
+        filenames = parse_filenames(added_lines)
+        commit_sha = subprocess.check_output(
+            ["git", "rev-parse", "HEAD"],
+            text=True
+        ).strip()
+        if not filenames:
+            print("No filenames found in changes")
+            shas = parse_sha256_digest(added_lines)
+            if not shas:
+                print("No sha256Digest found in changes")
+                return
+            else:
+                for sha in shas:
+                    filename, wheel_url = get_file_info_by_sha(sha)
+                    tag_name, release_title, version = generate_tag_and_title(filename)
+                    release_id, release_body, asset_id = get_release_info(tag_name)
+                    update_release_body(release_id, commit_sha, release_body, sha, tag_name) if release_id else None
+                    update_release_asset(wheel_url, asset_id, release_id) if release_id else None
+
+        print(f"Found {len(filenames)} files to process")
+        # Process each filename
+        for filename in filenames:
+            print(f"\nProcessing {filename}...")
+
+            # Get extension info from index.json
+            extension_info = get_extension_info(filename)
+            if not extension_info:
+                print(f"Could not get extension information for {filename}, skipping...")
+                continue
+
+            try:
+                tag_name, release_title, version = generate_tag_and_title(filename)
+                # Check if tag already exists
+                if check_tag_exists(tag_name):
+                    print(f"Tag {tag_name} already exists, skipping...")
+                    continue
+
+                # Try to get history notes from source code first
+                print(f"Getting history notes from source code...")
+                extension_name = re.match(r"^(.*?)[-_]\d+\.\d+\.\d+", filename).group(1)
+                history_note = get_history_note_from_source(version, extension_name)
+
+                # If no notes found in source code, try wheel package
+                if "No history notes found" in history_note:
+                    print(f"No history notes found in source code, trying wheel package...")
+                    history_note = get_history_note(extension_info["downloadUrl"], version)
```

Comment:
```suggestion
                    history_note = get_history_note_from_wheel(extension_info["downloadUrl"], version)
```
if release notes can be parsed from wheel always, we can directly ignore `get_history_note_from_source` above?

One thing about `get_history_note_from_source` is, some module, like `virtual-network-manager`, its module name is different from its source folder name in cli extension, which is `network-manager`, therefore it might always fail in `get_history_note_from_source`.

Reply:
Not sure if `get_history_note_from_wheel` can always work. But I'm OK we combine these two.

Reply:
Sometimes the HISTORY.rst file is not included in the wheel package, so it can only be read from the source code. Reading from the source code can also save time.